### ISSUE TITLE
linux-capture: ambiguous behavior when windows are not present/closed

### ISSUE THREAD

**Issue by MaZderMind** [obsproject/obs-studio#3040]
## Platform
Operating system and version: Ubuntu 19.10 (eoan)
OBS Studio version: `25.0.8` as well as `git master`

## Expected Behavior
 - Add Two Window-Capture (XComposite) Sources to capture two Browser-Windows
 - Close one of the Browser-Windows
 - The Source for the now-closed Window should stay at the same size
 - It should show some a Placeholder that the desired Window is not present
 - When the Window becomes available again (through re-opening the closed window), the new window is accepted again

## Current Behavior
 - When the Target-Window is closed, the Source switches to any other Window out of the same Window-Class, for example any open Browser-Window.
 - When the new Window has a Site different from the first, the Source even changes its Site in the Scene
 - When the Window is opened again, OBS does not recognize it and stays on the newly selected Window

## Steps to Reproduce
 - On Linux: Open Two Windows
  (I tested with Google Chrome Windows but it is reproducible with any Program)
 - Add Two Window-Capture (XComposite) Sources to capture the two Browser-Windows
 - Close one of the Browser-Windows
A Video capturing the Problem can be found here: https://mazdermind.de/dl/github/obs/3040/1.mp4

## Additional information
Reading the Source it seems that the Window-Selection-Code seems to solve this kind of problem by selecting Windows by Window-Class, when no Window match the exact Title. From an End-User perspective this leads towards ambiguous and unexpected behavior, because OBS changes the Scene and "randomly" selects another window, if there is one. If none is present, this does not happen.

I'm working on this and will try to provide a PR. I'm still opening this as a Bug to be open to a discussion and ideas.

**Comment by MaZderMind** [comment]
While working on this I found two related bugs in the same Code-Area:
 1. It is not possible work with two Windows that have the same name, because the window-selection does not try to find the Window by the Window-ID first but only looks at the Name
 2. When one Scene contains multiple Window-Captures of the same Window and the window is closed, `XCompcap::processEvents` obviously only receives one `DestroyNotify` and adds the window only once to the `changedWindows` collection. The first Source calls `XCompcap::windowWasReconfigured` which `erase`s the changed Window from the `changedWindows` collection, so that the second call (from the second source) to `XCompcap::windowWasReconfigured` does not find a change and returns `false`, thus leading to the source becoming broken and logging errors to the console:
```
error: X Error: BadWindow (invalid Window parameter), Major opcode: BadWindow (invalid Window parameter), Minor opcode: 0, Serial: 7855
```

**Comment by MaZderMind** [comment]
fixed in #3047
